src/bin/main.rs

```diff
@@ -287,6 +287,10 @@ fn execute(opts: &Options) -> FmtResult<Summary> {
             }
 
             let mut error_summary = Summary::default();
+            if let Some(mode) = options.write_mode {
+                error_summary.add_write_mode(mode)
+            }
+
```

Comment:
I thought I suggested a refactoring to avoid saving the write mode in the error summary, but maybe that was on another PR since I can't find any trace of it.

I think that by moving some code around in the main function so that we know the write mode (i.e., process the opts outside of `execute`), then we don't need to save it into the summary.

Reply:
> I thought I suggested a refactoring to avoid saving the write mode in the error summary, but maybe that was on another PR since I can't find any trace of it.

No worries.

> I think that by moving some code around in the main function so that we know the write mode (i.e., process the opts outside of execute), then we don't need to save it into the summary.

I think that's a good idea. I wanted to introduce a minimally invasive change with this PR, but if exit codes are gonna change, so should this.